lib/public_include/fty_shm.h

```diff
@@ -77,9 +88,21 @@ class shmMetrics
     // If you use this, DO NOT DELETE the fty_proto_t. It will be take
     // care by the shmlMetrics's destructor.
     //  (same warning if you access to it using iterator)
+
+    // @param index : index of the m_metricsVector
+    // @return : the address of the indexed element
```

Comment:
comment: not the address... + say if the user must delete the returned fty_proto_t* or not (is it a copy or a ptr referencing internal structure of the object)

Reply:
it's explained on top of the class, user must leave fty_proto_t without deleting.